### PR TITLE
Allow for non-existant anchor node ids when creating labels

### PR DESCRIPTION
**Pull request by kmontemayor2-sc**
Instead of https://github.com/Snapchat/GiGL/pull/78 let's go with this approach.

Now, if there isa. positive node id with no corresponding hard negative labels, the nodes negative labels will be padded with our padding node, which is later stripped out.

### CODE REVIEW

File: python/gigl/utils/data_splitters.py [hunk 1]
```diff
@@ -418,7 +435,10 @@ def _get_padded_labels(anchor_node_ids: torch.Tensor, topo: Topology) -> torch.T
     labels = torch.where(
         mask, torch.full_like(ranges, PADDING_NODE.item()), indices[ranges]
     )
-    return labels
+    labels = torch.cat(
+        [labels, torch.ones(extra_nodes_to_pad, max_range) * PADDING_NODE], dim=0
+    )
+    return labels.to(dtype=torch.int64)
```
Comment (mkolodner-sc) [inline python/gigl/utils/data_splitters.py:441]: why do we need `.to(dtype=torch.int64)` here? What is the dtype prior to this being called?

File: python/tests/unit/utils/data_splitters_test.py [hunk 2]
```diff
@@ -614,16 +616,20 @@ def test_get_labels_for_anchor_nodes_heterogeneous_positive_and_negative(self):
...
             dataset=ds,
             node_ids=node_ids,
             positive_label_edge_type=a_to_b,
             negative_label_edge_type=a_to_c,
         )
 
-        expected_positive = torch.tensor([[10, 11], [12, 13]])
-        expected_negative = torch.tensor([[20, -1], [30, 50]])
+        expected_positive = torch.tensor(
+            [[10, 11], [12, 13], [10, -1]], dtype=torch.int64
```
Comment (mkolodner-sc) [inline python/tests/unit/utils/data_splitters_test.py:628]: Would this be a [3x2] tensor, instead of the [2 x 3] tensor we should be expecting for edges?

File: python/tests/unit/utils/data_splitters_test.py [hunk 3]
```diff
@@ -614,16 +616,20 @@ def test_get_labels_for_anchor_nodes_heterogeneous_positive_and_negative(self):
...
+        )
+        expected_negative = torch.tensor(
```
Comment (mkolodner-sc) [inline python/tests/unit/utils/data_splitters_test.py:630]: Can you provide some context for how can we have these anchor nodes for our negative edges if our node_ids is [10, 11, 13]?

File: python/tests/unit/utils/data_splitters_test.py (hunk 2 above)
Comment (kmontemayor2-sc) [reply python/tests/unit/utils/data_splitters_test.py:628]: no? we expect to return `[num anchor nodes x num labels]` and we have three anchor nodes in this case

File: python/tests/unit/utils/data_splitters_test.py (hunk 3 above)
Comment (kmontemayor2-sc) [reply python/tests/unit/utils/data_splitters_test.py:630]: these are the other side of the edges, so we have `10 -> {10, 11}` as a positive label, and then our first set of labels (for node `10`) is `[10, 11]`.

File: python/gigl/utils/data_splitters.py (hunk 1 above)
Comment (kmontemayor2-sc) [reply python/gigl/utils/data_splitters.py]: it's sometimes float32... though maybe it's just testing?

The """problem""" here is the cat may convert everything to float if one part of it is float. Fixed now.

```
>>> torch.cat([torch.tensor([1, 2], dtype=torch.float32), torch.tensor([3], dtype=torch.int64)])
tensor([1., 2., 3.])
>>> torch.cat([torch.tensor([1, 2], dtype=torch.int64), torch.tensor([3], dtype=torch.float32)])
tensor([1., 2., 3.])
>>> torch.cat([torch.tensor([1, 2], dtype=torch.int64), torch.tensor([])])
tensor([1., 2.])
>>> torch.cat([torch.tensor([1, 2], dtype=torch.int64), torch.tensor([])])
tensor([1., 2.])
```

File: python/tests/unit/utils/data_splitters_test.py (hunk 3 above)
Comment (mkolodner-sc) [reply python/tests/unit/utils/data_splitters_test.py:630]: Awesome, thanks!

**Review by mkolodner-sc — APPROVED** [review]
Thanks Kyle! This LGTM provided tests pass :)
